styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -322,7 +308,8 @@ public void create(Environment environment) {
     closer.register(executorCloser("event-consumer", eventConsumerExecutor));
 
     final Stats stats = statsFactory.apply(environment);
-    final Storage storage = instrument(Storage.class, storageFactory.apply(environment), stats, time);
+    final Storage storage = instrument(Storage.class,
```

Comment:
Looks like storage is getting instrumented / wrapped in a proxy twice here

Reply:
Why?

Reply:
Both `instrument` and `new MeteredStorageProxy`

Reply:
Ah, `instrument` is a static method on the abstract class `MeteredProxy` which takes the concrete subclass as a parameter. That's a bit confusing to me.

Reply:
I agree. I have an idea how to fix. :D

Reply:
I took the liberty of pushing a commit to fix this =)